### PR TITLE
added first version of file extension excluder

### PR DESCRIPTION
**Pull request by phumberdroz**
### What does this implement/fix? Explain your changes.

 #224 
### Does this close any currently open issues?

 #224
### Any other comments?

this is a first version  excluding certian fileextension

things to be done:
- [x] Better Error / Warning Handler
- [x] Better Config title and description
- [ ] Add tests
### Checklist

Check all those that are applicable and complete.
- [ ] Merged with latest `master` branch
- [ ] Added examples for testing to [examples/ directory](examples/)
- [ ] Travis CI passes (Mac support)
- [ ] AppVeyor passes (Windows support)

### CODE REVIEW

File: src/beautify.coffee [hunk 1]
```diff
@@ -157,6 +157,19 @@ beautify = ({editor, onSave}) ->
     editedFilePath = editor.getPath()
 
 
+    # Get file extension everything after the first dot of the filename is considered the file extension
+    editedFileExtension = editor.getTitle().substr(editor.getTitle().indexOf("."))
+    # Get the extensions from the config
+    Extensions = atom.config.get('atom-beautify.general.beautifyFileExtensions').split " "
+    # console.log editedFileExtension
```
Comment (Glavin001) [inline src/beautify.coffee]: Delete commented out `console.log` statements.

File: src/beautify.coffee [hunk 2]
```diff
@@ -157,6 +157,19 @@ beautify = ({editor, onSave}) ->
     editedFilePath = editor.getPath()
 
 
+    # Get file extension everything after the first dot of the filename is considered the file extension
+    editedFileExtension = editor.getTitle().substr(editor.getTitle().indexOf("."))
+    # Get the extensions from the config
+    Extensions = atom.config.get('atom-beautify.general.beautifyFileExtensions').split " "
```
Comment (Glavin001) [inline src/beautify.coffee]: Change `Extensions` to `excludedFileExtensions` (or something more meaningful).

File: src/config.coffee [hunk 3]
```diff
@@ -32,6 +32,11 @@ module.exports = {
         type : 'boolean'
         default : true
         description : "When beautifying on save, use the entire file, even if there is selected text in the editor. Important: The `beautify on save` option for the specific language must be enabled for this to be applicable. This option is not `beautify on save`."
+      beautifyFileExtensions :
```
Comment (Glavin001) [inline src/config.coffee]: Change `beautifyFileExtensions` to `excludeFileExtensions` or something more accurate.

File: src/config.coffee [hunk 4]
```diff
@@ -32,6 +32,11 @@ module.exports = {
...
+        title: "Do not Beautify these extensions"
```
Comment (Glavin001) [inline src/config.coffee]: Title should resemble the key. So `beautifyFileExtensions` would have a title like `Beautify File Extensions`. It is useful when capitalization of certain words will not be predicted programmatically.
Any more information should be stored within the `description`, as you have done below.

File: src/config.coffee [hunk 5]
```diff
@@ -32,6 +32,11 @@ module.exports = {
...
+        type : 'string'
```
Comment (Glavin001) [inline src/config.coffee]: Change type `string` to `array`. See https://atom.io/docs/api/v1.6.0/Config for details.
For example:
```coffeescript
config:
  excludeFileExtensions:
    type: 'array'
    default: ['.min.js', '.min.css']
    items:
      type: 'string'
```

**Comment by phumberdroz** [comment]
@Glavin001 do you have a better way of notification handling somewhere in the code? I find the red error message a little rude to be honest. 

The atom info notification would be sufficient I think.

`atom.notifications.addInfo("This File does not get beautified", { description: "  Check Atom-Beautify General Settings" })`

**Comment by Glavin001** [comment]
> do you have a better way of notification handling somewhere in the code? I find the red error message a little rude to be honest.

I use the built-in Atom notifications. I think `addInfo` may be more appropriate here and less invasive 😄 .

> also on a right click beautify file it beautifies it again should I prevent this ?

You likely need to modify `beautifyFilePath`: https://github.com/phumberdroz/atom-beautify/blob/4c5f24b44131b4518633db5e2d69fd143a0a535b/src/beautify.coffee#L199
It is used for both `beautifyDirectory`: https://github.com/phumberdroz/atom-beautify/blob/4c5f24b44131b4518633db5e2d69fd143a0a535b/src/beautify.coffee#L286
And `beautifyFile`: https://github.com/phumberdroz/atom-beautify/blob/4c5f24b44131b4518633db5e2d69fd143a0a535b/src/beautify.coffee#L257

**Comment by phumberdroz** [comment]
will finish this one by end of the week so it is ready to get into new version do you have any deadline for PR merge ?

**Comment by Glavin001** [comment]
> do you have any deadline for PR merge ?

No rush 😄. The sooner the PR is ready for review and merge, the sooner I will publish a new release. However, if there are multiple PRs available to review and merge around the same time I may wait a day or so such that they can all join in on the next release. These are the ones I'd like to see in next release: https://github.com/Glavin001/atom-beautify/pulls?q=is%3Apr+is%3Aopen+label%3Ain-progress
Thank you!

**Comment by phumberdroz** [comment]
This should work now.

File: src/beautify.coffee [hunk 6]
```diff
@@ -187,6 +200,20 @@ beautify = ({editor, onSave}) ->
...
   logger.verbose('beautifyFilePath', filePath)
 
+  # Check if fileExtension is in excludeFileExtensions and then abort and show info
+  editedFileExtension = filePath.substr(filePath.indexOf("."))
+
+  # Get the extensions from the config
+  excludeFileExtensions = atom.config.get('atom-beautify.general.excludeFileExtensions')
+
+
+  if editedFileExtension in excludeFileExtensions
```
Comment (Glavin001) [inline src/beautify.coffee]: See comments above.

File: src/beautify.coffee [hunk 7]
```diff
@@ -187,6 +200,20 @@ beautify = ({editor, onSave}) ->
 beautifyFilePath = (filePath, callback) ->
   logger.verbose('beautifyFilePath', filePath)
 
+  # Check if fileExtension is in excludeFileExtensions and then abort and show info
+  editedFileExtension = filePath.substr(filePath.indexOf("."))
```
Comment (Glavin001) [inline src/beautify.coffee]: See comments above.

**Pull request closed without merging**